### PR TITLE
Change jss version in peer dependencies (+ small fixes)

### PR DESCRIPTION
**Pull request by umidbekkarimov**
All changes described in commits

Fix for jsstyles/jss#249

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+test
+node_modules
+webpack.config.js
```
Comment (kof) [inline .eslintignore]: why?

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,4 @@ tmp
 *~
 lib
 dist
+.idea/
```
Comment (kof) [inline .gitignore]: I don't think we should add every editors config out there ...

File: package.json [hunk 3]
```diff
@@ -32,21 +32,23 @@
...
+    "jss": "^4.0.3",
     "opener": "^1.4.1",
-    "qunitjs": "1.15.0",
+    "qunitjs": "^1.23.1",
     "rimraf": "^2.4.3",
     "webpack": "^1.12.2"
   },
   "peerDependencies": {
-    "jss": ">= 3.0.0 < 4.0.0"
+    "jss": "4.x.x"
```
Comment (kof) [inline package.json]: so now jss 3 is not supported any more?

File: src/index.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ const regExp = /([A-Z])/g
  * @return {String}
  */
 function replace(str) {
-  return '-' + str.toLowerCase()
+  return `-${str.toLowerCase()}`
```
Comment (kof) [inline src/index.js]: why?

File: src/index.js [hunk 5]
```diff
@@ -30,18 +30,16 @@ function convertCase(style) {
...
+  if (Array.isArray(style)) {
+    // Handle rules like @font-face, which can have multiple styles in an array
+    for (let index = 0; index < style.length; index++) {
+      style[index] = convertCase(style[index])
     }
   }
+  else {
+    rule.style = convertCase(style)
+  }
 }
```
Comment (kof) [inline src/index.js]: I miss the actual change here ... seems just formatting, no?

File: test/index.js [hunk 6]
```diff
@@ -33,14 +33,14 @@ test('camel cased @font-face with array of styles', function () {
       },
     ],
   })
-  const expected =
+  var expected =
```
Comment (kof) [inline test/index.js]: why?

File: .eslintignore (hunk 1 above)
Comment (umidbekkarimov) [inline .eslintignore]: For webstorm, it gives lot of errors in these files/dirs

File: test/index.js (hunk 6 above)
Comment (kof) [inline test/index.js]: ah right, plugins are still on the old infrastructure

File: .eslintignore (hunk 1 above)
Comment (kof) [inline .eslintignore]: Then we should lint this file.

Comment (kof) [inline .eslintignore]: it's because npm run lint is not linting this file, but your webstorm is using it's own runner

**Comment by kof** [comment]
also take a look at https://github.com/jsstyles/jss/issues/219

File: src/index.js (hunk 5 above)
Comment (umidbekkarimov) [inline src/index.js]: It's fix for eslint error `arrow-body-style`

File: src/index.js (hunk 4 above)
Comment (umidbekkarimov) [inline src/index.js]: And it's fix for eslint error `prefer-template`

File: .eslintignore (hunk 1 above)
Comment (umidbekkarimov) [inline .eslintignore]: > Then we should lint this file.

`test/index.js` should be ignored anyway, or have another .eslintrc file

**Pull request closed without merging**